lib/writer2.c

```diff
@@ -208,13 +208,19 @@ ktxTexture2_setImageFromStream(ktxTexture2* This, ktx_uint32_t level,
     if (!This->pData)
         return KTX_INVALID_OPERATION;
 
-    result = ktxTexture_GetImageOffset(ktxTexture(This),
-                                       level, layer, faceSlice,
-                                       &imageByteOffset);
-    if (result != KTX_SUCCESS)
-       return result;
-
-    imageByteLength = ktxTexture_GetImageSize(ktxTexture(This), level);
+    if (faceSlice == (ktx_uint32_t)(-1)) {
```

Comment:
Since this does not change the type of the faceSlice parameter, which I agree with, let's document this as specify UINT_MAX (i.e. the value of -1 cast to uint32_t) instead of -1 to set the entire level. Consider making a define to be used, e.g. KTX_ENTIRE_LEVEL.

Reply:
done, I did it that way to avoid including limits.h